doc/feature/spikeReport.md

```diff
@@ -0,0 +1,450 @@
+  can't be used for that!).
+  Returning a plain std object removes the information about the time window.
+  Even if a struct is used it implies that we will have to add another
+  function to update the internal state for doing it without copying any
+  data.
+
+In any case, from the average user's point of view it doesn't matter which
+option is chosen as long as the container models a STL-like container and it's
+possible to do binary searches on it.
+
+### 2: Why Spikes is a read-only class?
+
+_Resolved: Yes_
+
+By making it read-only, the implementation can share the same storage between
+multiple containers that refer to the same source but with different ranges.
+See also issues #1 and #2. For writing spikes the user can use a std::vector,
+whose interface doesn't differ very much from a potential write interface in
+Spikes.
+
+### 3: Why the access time of the spikes container is amortized constant time instead of constant time?
+
+_Resolved: Yes_
+
+The rationale for amortized time is that it allows the implementation to use
+data structures different from a single flat memory buffer per container. With
+special data structures it's possible to implement zero-copy optimizations,
+smarter internal storage and even internal threading while still ensuring
+const-correctness. For example, imagine we have some client code that uses
+OpenMP to speed up the processing of a report:
+
+    #pragma omp parallel for shared(report)
+    for (int i = 0; i != slices; ++i)
+    {
+        auto slice = getTimeSlice(i);
+        const brion::Spikes& spikes = report.getSpikes(slice.first,
+                                                       slice.second);
+        ... // Do something
+    }
+
+With strict constant time, the previous code cannot return any reference to
+shared data from getSpikes because it may be relocated by subsequent calls from
+other threads, hence the only sane alternative is to return a copy of the data.
+
+By relaxing the time complexity requirements, it would be possible to use a
+lunchbox::lfVector for example.
+
+### 4: Can getSpikes return _const Spikes&_?
+
+_Resolved: Yes_
+
+No. That will imply that the container needs to be stored inside SpikeReport.
+As a side-effect this requires Spikes to be copyable and the copies need to be
+efficient, but thanks to issues #1 and #2 the implementation can provide
+cheap copies by sharing.
+
+### 4: Do you need a method to erase portions of the report?
+
+_Resolved: Partially_
+
+We don't have a use case for this yet, but it looks necessary for out-of-core
+processing of very large reports. The mail goal for this method in the original
+class was to clear the internal caching. Clearing the report can be very
+complex without requiring duplication of data. The compromise solution is to
+provide a method to clear the cache enterily. The implementation can easily
+keep Spike containers returned by getSpikes valid with smart pointers.
```

Comment:
Difficult to say. In compartment reports the size of the data in a time window is known. With the spike report you don't know. A window of [0, 10s] could be to big or too small it terms of total data. At the same time, in the CompartmentReport you read frame by frame, and your buffer is clearly a set of frames. What does the buffering mean in a SpikeReport if I set the buffer size to 42 and then I do getSpikes(10, 100)?

I don't consider the consistency between the two report APIs so critical because they are fundamentally different. A compartment report is a function f: R, N -> R (time x offset -> value), a spike report is f: R, R -> set((R, N)).